config.go

```diff
@@ -105,6 +106,13 @@ func (c *Config) Discover() error {
 		}
 	}
 
+	if len(c.Providers) == 0 {
+		return errors.New("no providers found")
```

Comment:
Thanks! Just a cosmetic note about the style.... I think it's more idiomatic (and inline with the code style used in TF) to write this as `return fmt.Errorf("no providers found")`